specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/batchInference.json

```diff
@@ -0,0 +1,1712 @@
+        "computeId": {
+          "description": "Compute artifact Identifier",
+          "type": "string"
+        },
+        "errorThreshold": {
+          "format": "int32",
+          "description": "Error threshold, if the error count for the entire input goes above this value,\r\nthe batch inference will be aborted. Range is [-1, int.MaxValue]\r\n-1 value indicates, ignore all failures during batch inference\r\nFor FileDataset count of file failures\r\nFor TabularDataset, this is the count of record failures",
+          "type": "integer"
+        },
+        "retrySettings": {
+          "$ref": "#/definitions/BatchRetrySettings"
+        },
+        "partitioningScheme": {
+          "$ref": "#/definitions/BatchPartitioningScheme"
+        },
+        "loggingLevel": {
+          "$ref": "#/definitions/BatchLoggingLevel"
+        },
+        "outputConfiguration": {
+          "$ref": "#/definitions/BatchOutputConfiguration"
+        }
+      }
+    },
+    "InteractionEndpoints": {
+      "description": "Interaction endpoints",
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/InteractionEndpoint"
+      }
+    },
+    "InteractionEndpoint": {
+      "description": "Interaction endpoint",
+      "properties": {
+        "jobEndpoint": {
+          "$ref": "#/definitions/JobEndpoint"
+        },
+        "link": {
+          "type": "string"
+        }
+      }
+    },
+    "JobOutput": {
+      "description": "Output of a job.",
+      "type": "object",
+      "properties": {
+        "datastoreId": {
+          "description": "Datastore Identifier",
+          "type": "string"
+        },
+        "path": {
+          "description": "Path inside the datastore",
+          "type": "string"
+        }
+      }
+    },
+    "JobEndpoint": {
+      "description": "Job endpoint.",
+      "enum": [
+        "Tracking",
+        "Studio",
+        "Grafana",
+        "Tensorboard"
+      ],
+      "type": "string"
+    },
+    "JobStatus": {
+      "description": "The status of a job.",
+      "enum": [
+        "NotStarted",
+        "Starting",
+        "Provisioning",
+        "Preparing",
+        "Queued",
+        "Running",
+        "Finalizing",
+        "CancelRequested",
+        "Completed",
+        "Failed",
+        "Canceled",
+        "NotResponding"
+      ],
+      "type": "string"
+    },
+    "BatchJob": {
+      "description": "Batch job inference settings.",
+      "type": "object",
+      "properties": {
+        "scaleSettings": {
+          "$ref": "#/definitions/DeploymentScaleSettings"
+        },
+        "datasetId": {
+          "description": "Identifier for input dataset.",
+          "type": "string"
+        },
+        "tags": {
+          "description": "Contains resource tags defined as key/value pairs.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "properties": {
```

Comment:
what is this?  Why does the properties of the resource (this definition) need another dynamic properties dictionary inside it?

Reply:
Top level does have properties, that is what this definition is referenced by

Reply:
Removing here as well.